tests/Aspire.Hosting.Tests/Python/AddPythonProjectTests.cs

```diff
@@ -0,0 +1,159 @@
+            Assert.Equal(Path.Join(projectDirectory, ".venv", "bin", "opentelemetry-instrument"), pythonProjectResource.Command);
+        }
+
+        Assert.Equal("--traces_exporter", commandArguments[0]);
+        Assert.Equal("otlp", commandArguments[1]);
+        Assert.Equal("--logs_exporter", commandArguments[2]);
+        Assert.Equal("console,otlp", commandArguments[3]);
+        Assert.Equal("--metrics_exporter", commandArguments[4]);
+        Assert.Equal("otlp", commandArguments[5]);
+        Assert.Equal(pythonExecutable, commandArguments[6]);
+        Assert.Equal(scriptName, commandArguments[7]);
+    }
+
+    [Fact]
+    public async Task AddPythonProjectWithScriptArgs_IncludesTheArguments()
+    {
+        using var builder = TestDistributedApplicationBuilder.Create();
+
+        var (projectDirectory, pythonExecutable, scriptName) = CreateTempPythonProject();
+
+        builder.AddPythonProject("pythonProject", projectDirectory, scriptName, scriptArgs: "test");
+
+        var app = builder.Build();
+        var appModel = app.Services.GetRequiredService<DistributedApplicationModel>();
+        var executableResources = appModel.GetExecutableResources();
+
+        var pythonProjectResource = Assert.Single(executableResources);
+
+        Assert.Equal("pythonProject", pythonProjectResource.Name);
+        Assert.Equal(projectDirectory, pythonProjectResource.WorkingDirectory);
+
+        if (OperatingSystem.IsWindows())
+        {
+            Assert.Equal(Path.Join(projectDirectory, ".venv", "Scripts", "python.exe"), pythonProjectResource.Command);
+        }
+        else
+        {
+            Assert.Equal(Path.Join(projectDirectory, ".venv", "bin", "python"), pythonProjectResource.Command);
+        }
+
+        var commandArguments = await ArgumentEvaluator.GetArgumentListAsync(pythonProjectResource);
+
+        Assert.Equal(scriptName, commandArguments[0]);
+        Assert.Equal("test", commandArguments[1]);
+    }
+
+    private static (string projectDirectory, string pythonExecutable, string scriptName) CreateTempPythonProject(bool instrument = false)
+    {
+        
+
+        var projectDirectory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+        Directory.CreateDirectory(projectDirectory);
+
+        // Create a fake virtual environment.
+        var virtualEnvDirectory = Path.Combine(projectDirectory, ".venv");
+        Directory.CreateDirectory(virtualEnvDirectory);
+
+        string pythonExecutable = Path.Combine(virtualEnvDirectory, OperatingSystem.IsWindows() ? "Scripts\\python.exe" : "bin/python");
+
+        if (OperatingSystem.IsWindows())
+        {
+            var scriptsDirectory = Path.Join(virtualEnvDirectory, "Scripts");
+            Directory.CreateDirectory(scriptsDirectory);
+
+            File.WriteAllText(pythonExecutable, "");
+
+            if(instrument)
+            {
+                File.WriteAllText(Path.Join(scriptsDirectory, "opentelemetry-instrument.exe"), "");
```

Comment:
This path can also be returned from the method, which would simplify the checks for this path.

Reply:
Changed this up a bit to support running real Python.